storage/innobase/xtrabackup/src/xtrabackup.cc

```diff
@@ -7375,6 +7589,10 @@ int main(int argc, char **argv)
 	}
 #endif
 
+	if (opt_check_privileges) {
```

Comment:
It is more reliable to put that check after reading server variables, since some queries are executed only if some specific option is enabled on server. E.g. `have_backup_locks` and `have_flush_engine_logs` control if `FLUSH NO_WRITE_TO_BINLOG ENGINE LOGS` and `LOCK BINLOG FOR BACKUP` are going to be executed.